windows/utils/win_get_os_version.yml

```diff
@@ -0,0 +1,70 @@
+# Copyright 2023 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+# Get guest OS version with major and minor build number.
+# Return:
+#   win_os_version_build: Windows OS version.
+#
+- name: "Initialize the Windows OS version and registry path"
+  ansible.builtin.set_fact:
+    win_os_version_build: ""
+    win_os_version_reg_path: ""
+
+- name: "Get the registry key path for getting OS version"
+  include_tasks: win_execute_cmd.yml
+  vars:
+    win_powershell_cmd: >-
+      (Get-ChildItem -Path
+      "HKLM:\\SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\Update\\TargetingInfo\\Installed\\*.OS.*").Name
+    win_execute_cmd_ignore_error: true
+
+- name: "Set fact of registry key path for getting OS version"
+  ansible.builtin.set_fact:
+    win_os_version_reg_path: "{{ win_powershell_cmd_output.stdout_lines[0] | replace('HKEY_LOCAL_MACHINE', 'HKLM:') }}"
+  when:
+    - win_powershell_cmd_output.failed is defined
+    - not win_powershell_cmd_output.failed
+    - win_powershell_cmd_output.stdout_lines is defined
+    - win_powershell_cmd_output.stdout_lines | length == 1
+
+- name: "Get Windows OS version"
+  when: win_os_version_reg_path
+  block:
+    - name: "Get registry key value for getting OS version"
+      include_tasks: win_execute_cmd.yml
+      vars:
+        win_powershell_cmd: >-
+          Get-ItemPropertyValue -Path "{{ win_os_version_reg_path }}" -Name Version
+        win_execute_cmd_ignore_error: true
+    - name: "Set fact of Windows OS version"
+      ansible.builtin.set_fact:
+        win_os_version_build: "{{ win_powershell_cmd_output.stdout_lines[0] }}"
+      when:
+        - win_powershell_cmd_output.failed is defined
+        - not win_powershell_cmd_output.failed
+        - win_powershell_cmd_output.stdout_lines is defined
+        - win_powershell_cmd_output.stdout_lines | length == 1
+
+- name: "Get Windows OS version"
+  when: not win_os_version_reg_path
+  block:
+    - name: "Get registry key value for getting OS version"
+      include_tasks: win_execute_cmd.yml
+      vars:
+        win_powershell_cmd: >-
+          $majorver = Get-ItemPropertyValue -Path "HKLM:\\SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion" -Name CurrentMajorVersionNumber;
+          $minorver = Get-ItemPropertyValue -Path "HKLM:\\SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion" -Name CurrentMinorVersionNumber;
+          $buildnum = (Get-ItemPropertyValue -Path "HKLM:\\SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion" -Name BuildLabEx) -match "\d{5}\.\d{4}";
+          Write-Host($majorver, $minorver, $matches[0] -join '.')
```

Comment:
What if no matches?

Reply:
And it seems the value of BuildLabEx is not the actual build number. 
E.g. for Windows 11 v23H2, the winver show the build number is 22631.2861, while the value of BuildLabEx is "22621.1.amd64fre.ni_release.220506-1250"